qiskit/synthesis/__init__.py

```diff
@@ -94,13 +96,18 @@
     synth_permutation_basic,
     synth_permutation_acg,
 )
-from .linear import synth_cnot_count_full_pmh, synth_cnot_depth_line_kms
+from .linear import (
+    synth_cnot_count_full_pmh,
+    synth_cnot_depth_line_kms,
+    synth_cz_depth_line_mr,
+)
```

Comment:
I think the best solution is to have another folder of `synthesis/linear_phase` since the cz synthesis is into cx and phase gates, and relies on the theory of phase polynomials.

Reply:
Moving and renaming `graysynth`, as well as deprecating `https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/synthesis/graysynth.py` will be done in a separate PR